### PR TITLE
Fix: Reset Captcha if createOrder request fails

### PR DESCRIPTION
**Pull request by kewitz**
Resolves: https://github.com/opencollective/opencollective/issues/5063

Demo:
https://www.loom.com/share/2249be658dcf476fa636496acb2adb75

### CODE REVIEW

File: components/Captcha.js [hunk 1]
```diff
@@ -67,7 +67,8 @@ ReCaptcha.propTypes = {
   onError: PropTypes.func,
 };
 
-const Captcha = ({ onVerify, provider, ...props }) => {
+// eslint-disable-next-line react/display-name
```
Comment (Betree) [inline components/Captcha.js:70]: I think we can do `Captcha.displayName = 'Captcha';` after the component to get rid of the warning

File: components/contribution-flow/index.js [hunk 2]
```diff
@@ -624,7 +634,10 @@ class ContributionFlow extends React.Component {
             return true;
           } else {
             const isCompleted = Boolean(noPaymentRequired || stepPayment);
-            if (isCompleted && stepPayment?.key === NEW_CREDIT_CARD_KEY) {
+            if (isCaptchaEnabled() && !stepProfile.captcha && !LoggedInUser) {
+              this.showError('Captcha is required.');
```
Comment (Betree) [inline components/contribution-flow/index.js:638]: With the latest API from react-intl it's now pretty simple to i18n this

```suggestion
              this.showError(intl.formatMessage({ defaultMessage: 'Captcha is required.' }));
```

File: components/contribution-flow/index.js [hunk 3]
```diff
@@ -831,6 +844,14 @@ class ContributionFlow extends React.Component {
                     disabledPaymentMethodTypes={this.props.disabledPaymentMethodTypes}
                     hideCreditCardPostalCode={this.props.hideCreditCardPostalCode}
                   />
+                  {isCaptchaEnabled() && currentStep.name === STEPS.PAYMENT && !LoggedInUser && (
```
Comment (Betree) [inline components/contribution-flow/index.js:847]: The last step of the crypto flow seems different. Might be more future-proof to just check that we're on the last step:

```suggestion
                  {isCaptchaEnabled() && !nextStep && !LoggedInUser && (
```

**Review by Betree — COMMENTED** [review]
It seems that some parts conflict with what I just merged for the additional fields stuff. It shouldn't be too bad, but ff to ping me if I can help

**Comment by kewitz** [comment]
Rebased and fixed following review notes.

**Pull request merged**